src/goto-instrument/count_eloc.cpp

```diff
@@ -142,3 +145,50 @@ void print_path_lengths(const goto_modelt &goto_model)
       ++n_reachable;
   std::cout << "Reachable instructions: " << n_reachable << "\n";
 }
+
+void print_global_state_size(const goto_modelt &goto_model)
+{
+  const namespacet ns(goto_model.symbol_table);
+
+  // if we have a linked object, only account for those that are used
+  // (slice-global-inits may have been used to avoid unnecessary initialization)
+  goto_functionst::function_mapt::const_iterator f_it =
+    goto_model.goto_functions.function_map.find(INITIALIZE_FUNCTION);
+  const bool has_initialize =
+    f_it != goto_model.goto_functions.function_map.end();
+  std::unordered_set<irep_idt> initialized;
+
+  if(has_initialize)
+  {
+    for(const auto &ins : f_it->second.body.instructions)
+    {
+      if(ins.is_assign())
+      {
+        const code_assignt &code_assign = to_code_assign(ins.code);
+        const symbol_exprt &symbol_expr = to_symbol_expr(code_assign.lhs());
+
+        initialized.insert(symbol_expr.get_identifier());
+      }
+    }
+  }
+
+  mp_integer total_size = 0;
+
+  for(const auto &symbol_entry : goto_model.symbol_table.symbols)
+  {
+    const symbolt &symbol = symbol_entry.second;
+    if(
+      symbol.is_type || symbol.is_macro || symbol.type.id() == ID_code ||
+      symbol.type.get_bool(ID_C_constant) ||
+      (has_initialize && initialized.find(symbol.name) == initialized.end()))
+    {
+      continue;
+    }
+
+    const mp_integer bits = pointer_offset_bits(symbol.type, ns);
+    if(bits > 0)
```

Comment:
It seems to me that if the total is too large, the next thing someone will ask is "what's taking up all of the space", so could we have, possibly at verbose or debug, something that prints out the size and the name?

Reply:
I did consider doing so, but now that's done in #2184 during symbolic execution.

Reply:
OK.